app/helpers/operatingsystems_helper.rb

```diff
@@ -8,52 +8,57 @@ def label(record)
 
   def icon(record, opts = {})
     return "" if record.blank? || record.name.blank?
-    family = case record.name
-             when /fedora/i
-               "Fedora"
-             when /ubuntu/i
-               "Ubuntu"
-             when /solaris|sunos/i
-               "Solaris"
-             when /darwin/i
-               "Darwin"
-             when /centos/i
-               "Centos"
-             when /scientific/i
-               "Scientific"
-             when /archlinux/i
-               "Archlinux"
-             when /altlinux/i
-               "Altlinux"
-             when /gentoo/i
-               "Gentoo"
-             when /SLC/i
-               "SLC"
-             when /FreeBSD/i
-               "Freebsd"
-             when /aix/i
-               "AIX"
-             when /Junos/i
-               "Junos"
-             when /VRP/i
-               "VRP"
-             when /OracleLinux/i
-               "OracleLinux"
-             when /CoreOS|ContainerLinux|Container Linux/i
-               "Coreos"
-             when /RancherOS/i
-               "Rancheros"
-             when /NXOS/i
-               "NXOS"
-             when /XenServer/i
-               "Xenserver"
-             else
-               return "" if record.family.blank?
-               record.family
-             end
-    return image_path(family + ".png") if opts[:path]
+    size = opts[:size] ||= '16x16'
+    name = case record.name
+           when /fedora/i
+             "fedora"
+           when /ubuntu/i
+             "ubuntu"
+           when /solaris|sunos/i
+             "stub/steelblue-s"
+           when /darwin/i
+             "stub/darkred-d"
+           when /centos/i
+             "centos"
+           when /scientific/i
+             "scientific"
+           when /archlinux/i
+             "archlinux"
+           when /altlinux/i
+             "stub/goldenrod-a"
+           when /gentoo/i
+             "gentoo"
+           when /SLC/i
+             "stub/blue-s"
+           when /FreeBSD/i
+             "freebsd"
+           when /aix/i
+             "stub/forestgreen-a"
+           when /Junos/i
+             "stub/darkblue-j"
+           when /VRP/i
+             "firebrick-h"
+           when /OracleLinux/i
+             "firebrick-o"
+           when /CoreOS|ContainerLinux|Container Linux/i
+             "coreos"
+           when /RancherOS/i
+             "rancheros"
+           when /NXOS/i
+             "stub/darkslateblue-n"
+           when /XenServer/i
+             "stub/black-x"
+           when /Puppet/i
```

Comment:
Is Puppet an operating system family?

Reply:
I have no idea why this is there. The icon was present however, probably to keep it as I don't want to change behavior in this PR?